services/order.py

```diff
@@ -0,0 +1,33 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order
+
+from django.db import transaction
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime
+) -> None:
+    order = Order.objects.create(username=username)
+    if date:
+        order.created_at = date
+    order.save()
```

Comment:
You're calling the `save()` method twice for the `order` object which is against the DRY principle. You should only call `save()` once after you've finished modifying the object. Here's how you can refactor your code: 

python
order = Order.objects.create(username=username)
if date:
    order.created_at = date
order.save()
```